### ISSUE TITLE
Tail emission coinbase outputs, 2 per block down to 1? Saving ~50 bytes per block. 

### ISSUE THREAD

**Issue by ghost** [aeonix/aeon#251]
Can this be made into one output? I estimate it will save 50 bytes per block. Some options I can think of

- Adding a new denomination 1.2 
- making different block rewards such as a repeating sequence (1,1,1,1,1,2) so that the amount adds to be the same
- randomly selected so that it sums up to the same amount but distributes across all the denominations so as not to have millions of outputs at the same denomination.

**Comment by ghost** [comment]
Empty blocks with one output only take up ~100 bytes as opposed to ~140 bytes with 2 outputs and ~256 bytes for current empty blocks with 5 outputs. Given that empty blocks compose about 5/6 of all blocks we could see chain size growth decrease by half or more when tail emission comes.

**Comment by stoffu** [comment]
I'm against your proposal for the following reasons:

1. saving of 40 bytes is too small to justify consensus rule change
2. non-empty blocks have fees in addition to the base reward, so the number of outputs will be larger than one anyway
3. eventually (I hope) Aeon will become more popular and non-empty blocks will be dominant (otherwise no one cares about compactness of obscure blockchain with mostly empty blocks), so optimizing for empty blocks is counterproductive
4. when Monero's P2Pool functionality is ported to Aeon eventually, coinbase rewards will be split among miners participating in P2Pool, so the number of outputs will be larger than one

**Comment by ghost** [comment]
Yes, that is good feedback. I guess it is not important enough like you say. It naively seemed like an simple improvement but if a consensus change like adding a denomination is too controversial then I would not like to pursue this issue further.

**Comment by iamsmooth** [comment]
I think it's plausible to just stop denominating coinbase outputs. For pools (where its all usually public anyway), the coinbase outputs get combined to make pool payouts (which are denominated) and for a solo miner they can either do the same, or make one additional transaction per mined block if they really want denominated coinbase outputs without linking. We'd probably have to do this for p2pool anyway.

**Comment by stoffu** [comment]
@iamsmooth 
> I think it's plausible to just stop denominating coinbase outputs.

Do you mean to combine the coinbase output (which is the sum of base reward and fees) into just one output, or generate one output for the base reward (1.2 with tail emission) and multiple denominated outputs for the fees? The former idea is problematic because the amount will vary across blocks and thus those outputs cannot form ring signatures with other outputs of different amounts.

> We'd probably have to do this for p2pool anyway.

I don't understand why we need to stop denominating for enabling p2pool.

**Comment by iamsmooth** [comment]
@stoffu 

> because the amount will vary across blocks and thus those outputs cannot form ring signatures with other outputs of different amounts

Yes that's right. The first spend would be transparent, as was the case when we had the odd mixture of denominated and non-denominated outputs before. Except in this case, solely for coinbase outputs. 

As I said, often transparent spends for coinbase don't matter because its pools which often publish their blocks and payouts anyway, and in the case where someone does care, they can make one additional transaction, at slightly increased cost but not all that much (because the non-transparent spend is small and cheap), to accomplish the same thing (i.e. non-denominated -> denominated -> spend)

> I don't understand why we need to stop denominating for enabling p2pool.

Not strictly required, but since the coinbase would be split into small pieces to pool participants, you would need many decimal places. This would make the coinbase payouts huge (N decimal places times M pool members of coinbase outputs). I'm not sure that is worthwhile. But could be done as long as the block size permits it, so again, not strictly necessary to break denominating.

As above, miners can choose whether to accept the non-transparency of mining being the visible source of funds, or make one extra transaction to denominate.

Monero doesn't have this issue because with CT, it can make multiple precise coinbase payments without needing so many denominated outputs.
